Add zucchini version to title of test.

andrewbenton opened the pull request:
Title of the zucchini test output was always "1" before.  Now it at least provides some useful output.  This will help in identifying what version is actually being used and resolved by the POM so that debugging and error reporting is more accurate.

andrewbenton committed:
Added version introspection to output report

Added capability of JAR'd versions to show the version that is being
used.  This will help with future error reporting, as it means that the
user will be able to accurately report the version of zucchini that is
having problems.  This does not require an API change.

andrewbenton committed:
Merge branch 'develop' of github.com:andrewbenton/zucchini into develop

dimaj commented on src/main/java/com/comcast/zucchini/AbstractZucchiniTest.java:
```diff
@@ -264,7 +264,13 @@ else if(result.isJsonObject()) {
+                cleanup(context);
+            }
+            catch(Throwable t) {
+                LOGGER.error("ERROR: {}", t);
```
since we have no implementation of cleanup, I don't think it's wise for us to catch exceptions that have been introduced by the client.

andrewbenton replied:
We're doing a similar catch for setup() and setupFormatter().  Do you think it would be better if I changed it to catch a RuntimeException? 

As a side effect of the runWith() method throwing, a failed test will actually not be registered as a  failed test, so tests that throw in a parallel run can actually cause a failing set of tests to still register as passing. 

One way that I could fix this is by reversing the logic for failed tests so that it only registers passing tests, but this current logic would also halt serial runs and prevent other contexts from running, as it would terminate the run() method.

andrewbenton committed:
Shutdown hook to singleton.  Error log for setup.

The ZucchiniShutdownHook has be converted into a singleton instance,
which makes it easier to access it through the whole environment.  Along
with this is added logging to the ZucchiniShutdownHook that will
register whenever there is a problem with the setup and cleanup
procedures.  This should help to keep reporting more accurate.

andrewbenton committed:
Upgraded version and title labeling

Updated the ZucchiniShutdownHook so that the version is always labeled,
regardless of whether it is in a testing phase or a packaged phase.  The
title is now set through the environment variable ZUCCHINI_REPORT_TITLE.
 The version is now set in version.properties, which is altered by the
 pom using the maven-resources-plugin.

dimaj commented on src/main/java/com/comcast/zucchini/ZucchiniShutdownHook.java:
```diff
@@ -78,5 +133,25 @@ public void run() {
+                sb.append(String.format("Cause[%3d] :: %s\n", idx++, cause));
+            }
+
+            System.out.println("Zucchini failed with the following errors:");
```
Please no sysouts

andrewbenton committed:
Replace sysout with log to error.
